### ISSUE TITLE
is this project alive?

### ISSUE THREAD

**Issue by seven1986** [saaskit/saaskit#90]
can I use it for .net core 2?

**Comment by tgourley** [comment]
Seeing how the last commit was 16 months ago, I would say that it is not alive.  But I am implementing this into a new .NET Core 2 project and the concepts are still relevant.

**Comment by joeaudette** [comment]
I'm using it in .net core 2 and it is working just fine for my purposes

**Comment by quartilesoftware** [comment]
@joeaudette I am *trying* to use it in .Net Core 2,0 but am encountering the same problem another user reported a few days ago on Gitter:   

"  _We can't use UsePerTenant in ASP.NET Core 2.0 any longer....
At least not for the AddAuthentication().AddCookie(..).AddOpenIdConnect(...) middleware_ "

Have you been able to successfully use authentication cookies *per tenant* in your .Net Core 2.0 application where host names are used in App Tenant Resolution - e.g. http://host1.myapp.com; http://host2.myapp.com and so on? If you have, can you please share how you did that?

**Comment by joeaudette** [comment]
@quartilesoftware it is true that you cannot branch the auth middleware per tenant anymore, I still use sasskit to branch some other middleware.
For auth I had to implement resolving of identityoptions and cookie options, and social auth options  per tenant, relevant classes here https://github.com/joeaudette/cloudscribe/tree/master/src/cloudscribe.Core.Identity

Take a look at my project, maybe it will meet your needs and save you a lot of work.

**Comment by quartilesoftware** [comment]
@joeaudette 

Thanks very much for your comment and the link to CloudScribe.

Now that I seem to be being forced to replace SaasKit because of issues such as this, I had already looked at the CloudScribe project, but I need to implement a database per tenant, and so far as I can see, CloudScribe offers only a shared database solution. Is that correct?

**Comment by joeaudette** [comment]
that is correct, we have no implementation for that. however, probably whatever solution you arrive at for doing it with standard asp.net core identity, the same or similar solution might be possible with cloudscribe by overriding some things. I think you would have to have a list of tenants and connection strings in a central place and then probably some fancy DI to inject different dbcontextoptions with different connection strings per tenant. Not saying it would be easy or trivial though. Conceptually, using NoDb you have different folder for data per tenant so that is like a different db per tenant in that data is isolated per tenant, but NoDb would not be suitable for large sites with lots of users.

**Comment by RickBlouch** [comment]
@quartilesoftware 

You need to do something similar to what someone created a [pull request](https://github.com/saaskit/saaskit/pull/96) for.  It's the new way to do what we used to be able to do via tenant middleware branching.  It's using the `IConfigureNamedOptions `interface to update the cookie options on every request based on values in the tenant context.  I am personally using `IOptionsMonitor` instead, but same concept.

As for database per tenant, nothing should be different there.  Store your connection string in the tenant context and access it wherever you need.

**Comment by quartilesoftware** [comment]
@RickBlouch 

Brilliant! That's exactly what I need. Thanks.
